internal/ingress/controller/store/backend_ssl.go

```diff
@@ -104,17 +104,19 @@ func (s *k8sStore) getPemCertificate(secretName string) (*ingress.SSLCert, error
 			return nil, fmt.Errorf("unexpected error creating SSL Cert: %v", err)
 		}
 
+		path, err := ssl.StoreSSLCertOnDisk(nsSecName, sslCert)
```

Comment:
Why are you doing this?
Only CA certificates should be stored on disk. Everything else is handled by Lua, dynamically.

Reply:
When I used the original version and I defined proxy-ssl-secret in my Ingress the result was a config file in which all of these: proxy_ssl_trusted_cert, proxy_ssl_certificate, proxy_ssl_certificate_key pointed to the same file on disk. It was my experience with the original version. I.e. it wanted to have the tls.crt and tls.key from a file on a disk originally.
If I understand the original code well it does the following: puts the client cert and key into sslCert, then if server CA is present then it first stores the sslCert (i.e. client key and cert) on disk first, then it appends the server CA to the file.
In order to preserve this function I kept the writing the client key and cert to disk in my code.  I tested it, it works. Both when there is only ca.crt in the proxy-ssl-secret, and also when there are additional tls.key and tls.crt.

Reply:
@aledbf I would like to ask, could you please help me in finding that lua script that is supposed to manage the client key and cert? I would like to see the whole picture (even though my experience with the original version showed that the controller configured nginx to take those from the CAFile from disk) Thank you!